stacks-common/src/util/hash.rs

```diff
@@ -533,8 +527,8 @@ where
     /// Get the Merkle root hash.
     /// will be all 0's if the tree is empty.
     pub fn root(&self) -> H {
-        if self.nodes.len() > 0 {
-            if self.nodes[self.nodes.len() - 1].len() > 0 {
+        if !self.nodes.is_empty() {
```

Comment:
Is there a way to get Clippy to skip this transformation?  Semantically speaking, I find it easier to reason about a an array bounds-check in terms of the number of items it contains (i.e. the domain of valid indexes), instead of a container-generic `is_empty()` function.

Reply:
There definitely is a way. But this is one that I would probably push back on as is_empty is exactly this check (len() > 0) and it makes sense to me to check if its empty especially given the comment "will be all 0's if the tree is empty".

Reply:
I agree with @jferrant -- in most cases, the goal is to check that a container is empty and handle it as such. In this case, for example, the code is explicitly handling the situation in which the tree is empty (hence, the invocations of `H::empty()`. Thinking about this in terms of bounds checks is confusing: this code isn't really concerned with "array bounds checks", it should be concerned with checking the whole container for emptiness. Further, usually boundary checking suggests that something non-rustic is happening anyways (as in this function). This would be better represented as:

```rust
    pub fn root(&self) -> H {
        let Some(top_level_nodes) = self.nodes.last() else { return H::empty() };
        let Some(root) = top_level_nodes.first() else { return H::empty() };
        root.clone()
    }
```